src/test/test_snap_mapper.cc

```diff
@@ -573,6 +592,48 @@ class MapperVerifier {
       }
       hoids.clear();
     }
+    return ret;
+  }
+
+  // must be called with lock held to protect access to
+  // snap_to_hobject and hobject_to_snap in __trim_snap
+  // will keep trimming until reaching max_count or failing a call to __trim_snap()
+  void trim_snap_force(snapid_t           snapid,
+		       unsigned           max_count,
+		       vector<hobject_t>& out) {
+    int               guard = 1000;
+    vector<hobject_t> tmp;
+    unsigned          prev_size = 0;
+    while (tmp.size() < max_count) {
+      unsigned req_size = max_count - tmp.size();
+      // each call adds more objects into the tmp vector
+      __trim_snap(snapid, req_size, tmp);
+      if (prev_size < tmp.size()) {
+	prev_size = tmp.size();
+      }
+      else{
+	// the tmp vector size was not increased in the last call
+	// which means we were unable to find anything to trim
+	break;
+      }
+      ceph_assert(--guard > 0);
+    }
+    out.insert(out.end(), tmp.begin(), tmp.end());
+  }
+
+  void trim_snap() {
+    std::lock_guard l{lock};
+    if (snap_to_hobject.empty())
+      return;
+
```

Comment:
nit: unneeded blank – [`Minimize use of vertical whitespace`](https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace).